lib/_stream_writable.js

```diff
@@ -429,13 +429,11 @@ function onwriteError(stream, state, sync, er, cb) {
     // This can emit finish, and it will always happen
     // after error
     process.nextTick(finishMaybe, stream, state);
-    stream._writableState.errorEmitted = true;
     errorOrDestroy(stream, er);
   } else {
     // The caller expect this to happen before if
     // it is async
     cb(er);
-    stream._writableState.errorEmitted = true;
```

Comment:
This is removed on purpose. With this here the error from `destroy(err)` does not fully propagate to `on('error', ...)`.

Reply:
Seems breaking though 🤨

Reply:
hmm, you might be right. This needs https://github.com/nodejs/node/pull/29058.